src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -464,15 +465,18 @@ protected override IEnumerable<ObjectTypeProperty> BuildProperties()
             var propertiesFromSpec = GetParentPropertyDeclarationNames();
             var existingProperties = GetParentPropertySerializedNames();
 
-            foreach (var property in UpdateInputModelProperties())
+            foreach (var model in InputModel.GetSelfAndBaseModels())
```

Comment:
Is it possible to consolidate the behavior between MPG and DPG?
Since we are taking all the properties from the entire hierarchy, do we still need to fill in all the properties during CodeModelConverter?
If not, can we do the same thing for DPG?

Reply:
Never mind, we still need the composition properties since they don't exist in the hierarchy.

And we can consolidate MPG and DPG further, just want to avoid too many diffs in this PR.

Will approve after regen is good.

Reply:
Regen pipeline has been fixed, triggered a run for this: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3850400&view=results